app/controllers/products_controller.rb

```diff
@@ -0,0 +1,40 @@
+class ProductsController < ApplicationController
+  def new
+    @header = "Add A New Product"
+    @product = Product.new
+    @action = "create"
+    @method = :post
+  end
+  def edit
+    @header = "Edit Product"
+    id = params[:id]
+    @product = Product.find(id)
+    @action = "update"
+    @method = :patch
+    render "new"
+  end
+  def create
+    id = params[:vendor_id]
+    vendor = Vendor.find(id)
+    vendor.products.create(product_params)
+    redirect_to vendor_path(id)
+  end
+  def update
+    id = params[:vendor_id]
+    vendor = Vendor.find(id)
+    product = vendor.products.find(params[:id])
```

Comment:
I'm curious why you decided to go through the vendor to retrieve the product. Would there ever be a scenario where this product was not already associated with this vendor?